pythonforandroid/recipes/libffi/__init__.py

```diff
@@ -37,46 +36,10 @@ def build_arch(self, arch):
                     '--prefix=' + self.get_build_dir(arch.arch),
                     '--disable-builddir',
                     '--enable-shared', _env=env)
-            # '--with-sysroot={}'.format(self.ctx.ndk_platform),
-            # '--target={}'.format(arch.toolchain_prefix),
 
-            # ndk 15 introduces unified headers required --sysroot and
-            # -isysroot for libraries and headers. libtool's head explodes
-            # trying to weave them into it's own magic. The result is a link
-            # failure trying to link libc. We call make to compile the bits
-            # and manually link...
+            shprint(sh.make, '-j5', 'libffi.la', _env=env)
 
-            try:
-                shprint(sh.make, '-j5', 'libffi.la', _env=env)
-            except sh.ErrorReturnCode_2:
-                info("make libffi.la failed as expected")
-            cc = sh.Command(env['CC'].split()[0])
-            cflags = env['CC'].split()[1:]
             host_build = self.get_build_dir(arch.arch)
-
-            arch_flags = ''
```

Comment:
taking into account, the @AndreMiras comment about `lld` (https://github.com/kivy/python-for-android/pull/1744#pullrequestreview-215858810):

should be this block of code restored and made conditional? or is fixed by some of the other modification you did in this pr?
I suspect that if we don't use `lld`, we may still need this code..Am I right?

Reply:
Actually no, it doesn't work even with `lld`. Changing to `lld` will make it link successfully when compiling (for some reason), but there will still be runtime linker errors. The only way to fix the runtime linker errors was to fully fix the libffi compile. If for any reason we want to remove the use of `lld` in the future (i.e. remove the `-fuse-ld=lld` flag), it's completely safe to do so. All that was actually required to fix the libffi compile was changing `--sysroot {}` to `--sysroot={}`! :D

Reply:
Thanks for the reply, and yes I can confirm just skipping the `-fuse-ld=lld` was safe. I've tested it runtime and all seemed good